examples/tensorrt_llm/common/base_engine.py

```diff
@@ -368,3 +406,102 @@ async def async_llm_wrapper():
 
         self._llm_engine = None
         logger.info("Shutdown complete")
+
+    async def _get_remote_prefill_response(self, request):
+        prefill_request = copy.deepcopy(request)
+        prefill_request.sampling_params["max_tokens"] = 1
+        prefill_request.disaggregated_params = DisaggregatedParams(
+            request_type="context_only"
+        )
+
+        if self._prefill_client is None:
+            raise ValueError("Prefill client not initialized")
+
+        # TODO: Use smart KV router to determine which prefill worker to use.
+        ctx_responses = [
+            ctx_response
+            async for ctx_response in await self._prefill_client.round_robin(
+                prefill_request.model_dump_json()
+            )
+        ]
+        if len(ctx_responses) > 1:
+            raise ValueError(
+                "Prefill worker returned more than one response. This is currently not supported in remote prefill mode."
+            )
+        logger.debug(
+            f"Received response from prefill worker: {ctx_responses[0].data()}"
+        )
+        ctx_response_obj = TRTLLMWorkerResponse.model_validate_json(
+            ctx_responses[0].data()
+        )
+        ctx_response_obj.outputs = [
+            TRTLLMWorkerResponseOutput(**ctx_response_obj.outputs[0])
+        ]
+        assert ctx_response_obj.outputs[0].disaggregated_params is not None
+
+        return ctx_response_obj
+
+    async def generate(self, request: TRTLLMWorkerRequest):
+        if self._llm_engine is None:
+            raise RuntimeError("Engine not initialized")
+
+        if self._error_queue and self._error_queue.qsize() > 0:
+            error = self._error_queue.get()
+            raise error
+
+        self._ongoing_request_count += 1
+
+        try:
+            worker_inputs = request.tokens.tokens
+
+            disaggregated_params = (
+                DisaggregatedTypeConverter.to_llm_disaggregated_params(
+                    request.disaggregated_params
+                )
+            )
+
+            if self._remote_prefill and self._server_type == ServerType.GEN:
```

Comment:
question: do we need both of these conditions - would remote_prefill be sufficient?

Reply:
The ctx worker calls into the same function and we would not want it to keep calling remote prefill iteratively.